src/Source.cpp

```diff
@@ -335,6 +335,87 @@ void SourceUniformCylinder::setDescription() {
 	description = ss.str();
 }
 
+// ---------------------------------------------------------------------------
+SourceSNRDistribution::SourceSNRDistribution() :
+    R_earth(8.5*kpc), beta(3.53), Zg(0.3*kpc) {
+	set_frMax(8.5*kpc, 3.53);
+	set_fzMax(0.3*kpc);
+}
+
+SourceSNRDistribution::SourceSNRDistribution(double R_earth, double beta, double Zg) :
+    R_earth(R_earth), beta(beta), Zg(Zg) {
+	set_frMax(R_earth, beta);
+	set_fzMax(Zg);
+}
+
+void SourceSNRDistribution::prepareParticle(ParticleState& particle) const {
+  	Random &random = Random::instance();
+	double RPos;
+	while (true){
+		RPos = random.rand()*20.*kpc;
```

Comment:
There is no physics reason. It is only for efficiency using the JF12 field which is cut to zero at R=20 kpc. Probably it is better to just set the maximum values by default and make them variable. In that way the source class could be reused for other magnetic fields. 
I will fix that tomorrow.